core/lib/presentation/views/quick_search/quick_search_suggestion_list.dart

```diff
@@ -0,0 +1,526 @@
+    this.onSuggestionSelected,
+    this.suggestionsCallback,
+    this.itemBuilder,
+    this.scrollController,
+    this.decoration,
+    this.debounceDuration,
+    this.loadingBuilder,
+    this.noItemsFoundBuilder,
+    this.errorBuilder,
+    this.transitionBuilder,
+    this.animationDuration,
+    this.animationStart,
+    this.direction,
+    this.hideOnLoading,
+    this.hideOnEmpty,
+    this.hideOnError,
+    this.keepSuggestionsOnLoading,
+    this.minCharsForSuggestions,
+    this.listActionButton,
+    this.actionButtonBuilder,
+    this.buttonActionCallback,
+    this.buttonShowAllResult,
+    this.titleHeaderRecent,
+    this.itemRecentBuilder,
+    this.fetchRecentActionCallback,
+    this.onRecentSelected,
+    this.listActionPadding,
+    this.hideSuggestionsBox = false,
+    this.isDirectionRTL = false,
+    this.minInputLengthAutocomplete,
+    this.contactSuggestionBuilder,
+    this.contactSuggestionCallback,
+    this.onContactSuggestionSelected,
+  });
+
+  @override
+  QuickSearchSuggestionListState<T, P, R> createState() =>
+      QuickSearchSuggestionListState<T, P, R>();
+}
+
+class QuickSearchSuggestionListState<T, P, R>
+    extends State<QuickSearchSuggestionList<T, P, R>>
+    with SingleTickerProviderStateMixin {
+  Iterable<T>? _suggestions;
+  Iterable<R>? _recentItems;
+  Iterable<P>? _contacts;
+  late bool _suggestionsValid;
+  bool? _isLoading;
+  AnimationController? _animationController;
+  String? _lastTextValue;
+  late final ScrollController _scrollController =
+      widget.scrollController ?? ScrollController();
+  late final Debouncer<String> _deBouncerSuggestion;
+  late final StreamSubscription<String> _deBouncerSuggestionStreamSubscriptions;
+
+  Future<Iterable<R>?> _getListRecent(String queryString) async {
+    try {
+      return widget.fetchRecentActionCallback?.call(queryString);
+    } catch (e) {
+      logError('SuggestionsListState::_getRecent:Exception = $e');
+      return null;
+    }
+  }
+
+  Future<Iterable<P>?> _getListContact(String queryString) async {
+    try {
+      return widget.contactSuggestionCallback?.call(queryString);
+    } catch (e) {
+      logError('SuggestionsListState::_getListContact:Exception = $e');
+      return null;
+    }
+  }
+
+  Future<Iterable<T>?> _getListSuggestion(String queryString) async {
+    try {
+      return widget.suggestionsCallback?.call(queryString);
+    } catch (e) {
+      logError('SuggestionsListState::_getListSuggestion:Exception = $e');
+      return null;
+    }
+  }
+
+  Future<void> _handleDebounceTimeListener(String queryString) async {
+    log('QuickSearchSuggestionListState::_handleDebounceTimeListener:queryString = $queryString | minCharsForSuggestions = ${widget.minCharsForSuggestions}');
+    if (!mounted) return;
+
+    if (widget.minCharsForSuggestions != null &&
+        queryString.length <= widget.minCharsForSuggestions!) {
+      final recentItems = await _getListRecent(queryString);
```

Comment:
Will this method cause race condition? If yes, should we implement cancel token? E.g. 
- Method 1 is `_handleDebounceTimeListener('abc')`
- Method 2 is `_handleDebounceTimeListener('abc123')`
- Method 1 is called before method 2, but method 1 response after method 2 response

Reply:
Good catch. Fixed